@types/esquery/index.d.ts

```diff
@@ -1,5 +1,8 @@
 declare module "esquery" {
   import * as es from "estree";
-  declare function esquery(node: es.Node, selector: string): es.Node[];
+  declare function esquery<T extends es.Node = es.Node>(
```

Comment:
I disagree with (ab)using type parameters like this. Here the type parameter will always be used as an assertion. I think this is wrong and it would be better practice to use an actual assertion.

I made some comments on something similar [here](https://github.com/ReactiveX/rxjs/pull/4474#issuecomment-519279746).

I think explicit type parameters in the value space should be avoided wherever possible. They should only be needed in situations in which TypeScript is incapable of inferring a type and here inference is a non-issue - this is a straight-up type assertion.

Reply:
I haven't thought of it like this before, but it makes sense the way you put have put it 👍. Thanks for the insights!